domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2730,6 +2731,36 @@ public void streamGetWithFacetsAndFilter(TestContext context) throws IOException
     async.awaitSuccess();
   }
 
+  @Test
+  public void normalGetWithFacetsAndFilter(TestContext context) throws IOException, FieldException {
+    AtomicInteger objectCount = new AtomicInteger();
+    Async async = context.async();
+
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    List<FacetField> facets = new ArrayList<FacetField>();
+    facets.add(new FacetField("jsonb->>'edition'"));
+    facets.add(new FacetField("jsonb->>'title'"));
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=First edition");
+
+    postgresClient.get(MOCK_POLINES_TABLE, Poline.class, new String[]{"jsonb"}, wrapper, true, true, facets, context.asyncAssertSuccess(sr -> {
+      ResultInfo resultInfo = sr.getResultInfo();
+      context.assertEquals(3, resultInfo.getTotalRecords());
+      context.assertEquals(2, resultInfo.getFacets().size());
+      context.assertEquals("edition", resultInfo.getFacets().get(0).getType());
+      context.assertEquals(1, resultInfo.getFacets().get(0).getFacetValues().size());
+      context.assertEquals("First edition", resultInfo.getFacets().get(0).getFacetValues().get(0).getValue());
+      context.assertEquals(3, resultInfo.getFacets().get(0).getFacetValues().get(0).getCount());
+      context.assertEquals("title", resultInfo.getFacets().get(1).getType());
+      context.assertEquals(3, resultInfo.getFacets().get(1).getFacetValues().size());
+      context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(0).getCount());
+      context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(1).getCount());
+      context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(2).getCount());
+      async.complete();
+    }));
+    async.awaitSuccess();
```

Comment:
Async is not needed because there is only one async callback and that callback has a context.asyncAssertSuccess() handler that automatically awaits success.

Reply:
I'm probably going to ignore the remove async stuff in the future. Bottomline is that async covers all cases, while the others can only be used in special cases.